### PR TITLE
[sharktank] Fix which prefill logits IREE perplexity calculations are using

### PR DESCRIPTION
**Pull request by Alex-Vasile**
The current implementation had two issues:
1. IREE and eager perplexity calculations were using different prefill logits
2. Neither were taking into account the sequence length of a given prompt within the batch when grabbing a slice of the logits.

This PR changes the IREE perplexity calculation to use the correct prefill logits; each entry in the batch the slice corresponding to the final real token is being used to calculate perplexity.

Only fixing the IREE side, the eager side is untouched (and left incorrect) because of #1509 prevent rerunning to get new values. Leaving eager calc as is also allows for #1509 to be more easily fixed.

TODO:
- [x] Update perplexity scores in `baseline_perplexity_scores.json`

For: #1495

### CODE REVIEW

File: sharktank/tests/evaluate/baseline_perplexity_scores.json [hunk 1]
```diff
@@ -372,268 +372,268 @@
...
+      5.914172,
+      16.327044,
+      14.681435,
+      6.076033,
+      11.515375,
+      9.710052,
+      10.319663
     ],
-    "mean_perplexity": 14.629444
+    "mean_perplexity": 14.009884
```
Comment (Alex-Vasile) [inline sharktank/tests/evaluate/baseline_perplexity_scores.json:504]: Iree perplexity going down is great to see!

Comment (archana-ramalingam) [reply sharktank/tests/evaluate/baseline_perplexity_scores.json:504]: Yes, this is great!
